cms/djangoapps/contentstore/helpers.py

```diff
@@ -299,31 +330,41 @@ def import_staged_content_from_user_clipboard(parent_key: UsageKey, request) ->
             tags=user_clipboard.content.tags,
         )
 
-        # Now handle static files that need to go into Files & Uploads.
-        static_files = content_staging_api.get_staged_content_static_files(user_clipboard.content.id)
-        notices, substitutions = _import_files_into_course(
-            course_key=parent_key.context_key,
-            staged_content_id=user_clipboard.content.id,
-            static_files=static_files,
-            usage_key=new_xblock.scope_ids.usage_id,
-        )
-
-        # Rewrite the OLX's static asset references to point to the new
-        # locations for those assets. See _import_files_into_course for more
-        # info on why this is necessary.
-        if hasattr(new_xblock, 'data') and substitutions:
-            data_with_substitutions = new_xblock.data
-            for old_static_ref, new_static_ref in substitutions.items():
-                data_with_substitutions = data_with_substitutions.replace(
-                    old_static_ref,
-                    new_static_ref,
-                )
-            new_xblock.data = data_with_substitutions
-            store.update_item(new_xblock, request.user.id)
+        notices = _insert_static_files_into_downstream_xblock(new_xblock, user_clipboard.content.id, request)
 
     return new_xblock, notices
 
 
+def import_staged_content_for_library_sync(downstream_xblock: XBlock, lib_block: XBlock, request) -> StaticFileNotices:
+    """
+    Import the static assets from the library xblock to the downstream xblock
+    through staged content. Also updates the OLX references to point to the new
+    locations of those assets in the downstream course.
+
+    Does not deal with permissions or REST stuff - do that before calling this.
+
+    Returns a summary of changes made to static files in the destination
+    course.
+    """
+    if not content_staging_api:
+        raise RuntimeError("The required content_staging app is not installed")
+    staged_content = content_staging_api.stage_xblock_temporarily(lib_block, request.user.id)
+    if not staged_content:
+        # expired/error/loading
+        return StaticFileNotices()
+
+    store = modulestore()
+    with store.bulk_operations(downstream_xblock.context_key):
+        # Now handle static files that need to go into Files & Uploads.
+        # If the required files already exist, nothing will happen besides updating the olx.
+        try:
+            notices = _insert_static_files_into_downstream_xblock(downstream_xblock, staged_content.id, request)
+        finally:
+            staged_content.delete()
```

Comment:
```suggestion
    try:
        # Now handle static files that need to go into Files & Uploads.
        # If the required files already exist, nothing will happen besides updating the olx.
        with store.bulk_operations(downstream_xblock.context_key):
            notices = _insert_static_files_into_downstream_xblock(downstream_xblock, staged_content.id, request)
    finally:
        staged_content.delete()
```
You don't have to keep the modulestore bulk operation open while deleting the staged content, since staged content doesn't use modulestore.